feat(auth): add Yahoo as provider for iOS, Android and Web (#9443)
firebase/flutterfire: docs/auth/federated-auth.md

    // return await FirebaseAuth.instance.signInWithRedirect(githubProvider);
  }
  ```

## Yahoo

Ensure the "Yahoo" sign-in provider is enabled on the [Firebase Console](https://console.firebase.google.com/project/_/authentication/providers)
with an API Key and API Secret set. Also make sure your Firebase OAuth redirect URI (e.g. my-app-12345.firebaseapp.com/__/auth/handler) 

> is set as your Authorization callback URL in your app's settings page on your Yahoo app's config.

Reviewer: ```suggestion
is set as your Authorization callback URL in your app's settings page on your  app's Yahoo config.
```

* {iOS+}

>   Before you begin [configure Yahoo Login for iOS](/docs/auth/ios/yahoo-oauth#before_you_begin) and add the [custom URL schemes

Reviewer: ```suggestion
  Before you begin, [configure Yahoo Login for iOS](/docs/auth/ios/yahoo-oauth#before_you_begin) and add the [custom URL schemes
```

  to your Runner (step 1)](https://firebase.google.com/docs/auth/ios/yahoo-oauth#handle_the_sign-in_flow_with_the_firebase_sdk).

* {Android}

>   Before you begin [configure Yahoo Login for Android](/docs/auth/android/yahoo-oauth#before_you_begin).

Reviewer: ```suggestion
  Before you begin, [configure Yahoo Login for Android](/docs/auth/android/yahoo-oauth#before_you_begin).
```

    // return await FirebaseAuth.instance.signInWithRedirect(githubProvider);
  }
  ```

## Yahoo

Ensure the "Yahoo" sign-in provider is enabled on the [Firebase Console](https://console.firebase.google.com/project/_/authentication/providers)
with an API Key and API Secret set. Also make sure your Firebase OAuth redirect URI (e.g. my-app-12345.firebaseapp.com/__/auth/handler) 

> is set as your Authorization callback URL in your app's settings page on your app's Yahoo config.

Reviewer 2: ```suggestion
is set as a redirect URI in your app's Yahoo Developer Network configuration.
```

Ensure the "Yahoo" sign-in provider is enabled on the [Firebase Console](https://console.firebase.google.com/project/_/authentication/providers)
with an API Key and API Secret set. Also make sure your Firebase OAuth redirect URI (e.g. my-app-12345.firebaseapp.com/__/auth/handler) 
is set as your Authorization callback URL in your app's settings page on your app's Yahoo config.

> .

Reviewer 2: Stray period
